spec/system/provider_interface/manage_provider_user_permissions_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+RSpec.feature 'Managing provider user permissions' do
+  include DfESignInHelpers
+
+  scenario 'Provider manages permissions for users' do
+    given_i_am_a_provider_user_with_dfe_sign_in
+    and_the_provider_add_provider_users_feature_is_enabled
+    and_i_can_manage_applications_for_two_providers
+    and_i_can_manage_users_for_a_provider
+    and_i_sign_in_to_the_provider_interface
+
+    when_i_click_on_the_users_link
+    and_i_click_on_a_user
+    and_i_click_change_providers_and_permissions
+
+    then_i_see_providers_and_permissions
+    and_i_add_permission_to_manage_users_for_a_provider_user
+    then_i_can_see_the_new_permission_for_the_provider_user
+    and_i_click_change_providers_and_permissions
+
+    when_i_remove_a_permissions_from_a_provider_user
+    then_i_cant_see_the_permission_for_the_provider_user
+  end
+
+  def given_i_am_a_provider_user_with_dfe_sign_in
+    provider_exists_in_dfe_sign_in
+  end
+
+  def and_i_can_manage_applications_for_two_providers
+    provider_user_exists_in_apply_database
+    @provider_user = ProviderUser.find_by(dfe_sign_in_uid: 'DFE_SIGN_IN_UID')
+    @provider = Provider.find_by(code: 'ABC')
+    @another_provider = Provider.find_by(code: 'DEF')
+  end
+
+  def and_i_can_manage_users_for_a_provider
+    @provider_user.provider_permissions.find_by(provider: @provider).update(manage_users: true)
+  end
+
+  def and_the_provider_add_provider_users_feature_is_enabled
+    FeatureFlag.activate('provider_add_provider_users')
+  end
+
+  def when_i_click_on_the_users_link
+    click_on('Users')
+  end
+
+  def and_i_click_on_a_user
+    click_on(@provider.provider_users.first.full_name)
```

Comment:
Is this the current user themself that we're editing? Or are we creating a separate user somewhere?